### PR TITLE
[improve JENKINS-30107] print ip causing the failed login

### PR DESCRIPTION
**Pull request by lexxxel**
this PR addresses the problem described [here](https://issues.jenkins-ci.org/browse/JENKINS-30107).
By adding the ip address to the log, I'm able to use fail2ban to prevent brute force attacks, but of cause someone could also write a plugin for this.

### CODE REVIEW

File: core/src/main/java/jenkins/security/SecurityListener.java [hunk 1]
```diff
@@ -112,10 +112,10 @@ public static void fireLoggedIn(@Nonnull String username) {
     }
 
     /** @since 1.569 */
-    public static void fireFailedToLogIn(@Nonnull String username) {
-        LOGGER.log(Level.FINE, "failed to log in: {0}", username);
+    public static void fireFailedToLogIn(@Nonnull String username, @Nonnull String ip) {
```
Comment (jtnord) [inline core/src/main/java/jenkins/security/SecurityListener.java]: this breaks backwards compatability.

**Comment by jtnord** [comment]
unfortunately the changes as implemented breaks backwards compatibility as all security Listeners (can be defined in plugins) will need to be updated.

File: core/src/main/java/jenkins/security/SecurityListener.java [hunk 2]
```diff
@@ -71,7 +71,7 @@
      * Would be called after {@link #failedToAuthenticate}.
      * @param username the user
      */
-    protected abstract void failedToLogIn(@Nonnull String username);
+    protected abstract void failedToLogIn(@Nonnull String username, @Nonnull String ip);
```
Comment (jtnord) [inline core/src/main/java/jenkins/security/SecurityListener.java]: this breaks backwards compatability for all `SecurityListener` implementations.

File: core/src/main/java/hudson/security/AuthenticationProcessingFilter2.java [hunk 3]
```diff
@@ -103,7 +103,12 @@ protected void onUnsuccessfulAuthentication(HttpServletRequest request, HttpServ
         LOGGER.log(Level.FINE, "Login attempt failed", failed);
         Authentication auth = failed.getAuthentication();
         if (auth != null) {
-            SecurityListener.fireFailedToLogIn(auth.getName());
+        	//is client behind something?
+			String ipAddress = request.getHeader("X-FORWARDED-FOR");
```
Comment (daniel-beck) [inline core/src/main/java/hudson/security/AuthenticationProcessingFilter2.java]: This should be taken care of by the server. We don't use this for the `Cause.RemoteCause` either, for example:
https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/jenkins/model/ParameterizedJobMixIn.java#L258

See https://github.com/jenkinsci/winstone/pull/13

**Comment by daniel-beck** [comment]
I don't know wtf is going on here, but… no. Just, no.

Please file a new PR with only the relevant commits (addressing previous review comments) if you want to have this merged.

**Comment by lexxxel** [comment]
@daniel-beck I completly forgot about this PR.... what happend here?

**Comment by daniel-beck** [comment]
@lexxxel My guess is a merge gone wrong. Best to start over with a new branch.

**Pull request closed without merging**